astropy/cosmology/_src/core.py

```diff
@@ -203,19 +208,19 @@ def _register_cls(cls):
 
     # ---------------------------------------------------------------
 
-    def __post_init__(self):  # noqa: B027
+    def __post_init__(self) -> None:  # noqa: B027
         """Post-initialization, for subclasses to override if they need."""
 
     @property
     @abstractmethod
-    def is_flat(self):
+    def is_flat(self) -> bool:
         """Return bool; `True` if the cosmology is flat.
 
         This is abstract and must be defined in subclasses.
         """
         raise NotImplementedError("is_flat is not implemented")
 
-    def clone(self, *, meta=None, **kwargs):
+    def clone(self, *, meta: Mapping | None = None, **kwargs: Any) -> "Cosmology":
```

Comment:
`Mapping` is a generic type with two parameters.

Reply:
Yes. It should be `Mapping[Any, Any]`...